src/main/java/com/commercetools/sync/services/impl/CustomObjectServiceImpl.java

```diff
@@ -55,10 +62,22 @@ public CustomObjectServiceImpl(@Nonnull final CustomObjectSyncOptions syncOption
     public CompletionStage<Set<CustomObject<JsonNode>>> fetchMatchingCustomObjectByCompositeIdentifiers(
             @Nonnull final Set<CustomObjectCompositeIdentifier> identifiers) {
 
+        Set<CustomObjectCompositeIdentifier> filteredIdentifiers = identifiers.stream()
+                .filter(identifier ->
+                            StringUtils.isNotEmpty(identifier.getContainer())
+                        &&  StringUtils.isNotEmpty(identifier.getKey()))
+                .collect(Collectors.toSet());
+
+        if (filteredIdentifiers.size() == 0) {
+            return CompletableFuture.completedFuture(Collections.EMPTY_SET);
```

Comment:
```suggestion
            return CompletableFuture.completedFuture(Collections.emptySet());
```

Reply:
What's difference here?

Reply:
One is generified, the other one just returns Object as an generic type:
```
final CompletableFuture<Set> setCompletableFuture = CompletableFuture.completedFuture(Collections.EMPTY_SET);
final CompletableFuture<Set<CustomObject<JsonNode>>> setCompletableFuture = CompletableFuture.completedFuture(Collections.emptySet());
```

From the docs of `emptySet()`:
```
* Returns an empty set (immutable).  This set is serializable.
     * Unlike the like-named field, this method is parameterized.
     *
     * <p>This example illustrates the type-safe way to obtain an empty set:
     * <pre>
     *     Set&lt;String&gt; s = Collections.emptySet();
     * </pre>
```

Reply:
Done
https://github.com/commercetools/commercetools-sync-java/pull/557/commits/5d32468abc9b76bc3de823da8b56751be997c019